docs/source/coding_guidelines.md

```diff
@@ -0,0 +1,213 @@
+# Design and Coding Guidelines
+
+## Design Phase
+
+For non-trivial changes, please communicate your plans early on to reach maximum impact and to avoid duplicate work:
+
+* What are you planning to working on?
+* Why is it needed for the project?
+* Why is it needed in the repository?
+* Ask the rest of the team if they have already worked on it or thought about it. This helps to get to know about existing code fragments that can simplify your task, and integrate better with existing code.
+
+For non-trivial changes, we recommend to work with a team member to create a design for your work.
+Depending on the size of the task at hand, a design can be anywhere between a short descrption of
+your plan on Github, or a long document where you list your design options and invite feedback.
+
+Your design partner will later also be well positioned to review your PR because they are familiar with your plan.
+
+## Setting up your dev environment
+
+Please see the detailed instructions [here](developers.md).
+
+## Coding Guidelines
+
+### Naming
+
+Please follow the general Python rules for naming:
+
+* Variables, function and method names should use `snake_case` (lower case with under scores separating words)
+* Class names should follow `PascalCase` (first letter of each word capitalized).
```

Comment:
Maybe just mention that PEP8 must be followed?

Reply:
I'd rather spell it out

Reply:
Makes sense. But I'd add a link to PEP8, then.